corehq/form_processor/backends/sql/dbaccessors.py

```diff
@@ -628,13 +630,24 @@ def save_new_form(form):
             operation.form_id = form.form_id
 
         try:
+            db_names = [form.db]
+            tracked_objects = get_tracked_objects_to_create(form)
```

Comment:
instead of doing this which adds quite a bit of complexity I think you can just do this:
```python
db_names = {form.db}
additional_models = form.create_models
for model in additional_models:
    if if hasattr(model, 'is_saved) and model.is_saved():
        raise XFormSaveError(
            '{} {} has already been saved'.format(type(model) model.pk)
        )
    model.form_id = form.form_id
    if isinstance(model, PartitionedModel):
        db_names.add(model.db)
    else:
        db_names.add(router.db_for_write(type(model)))
```

This would replace the section above: https://github.com/dimagi/commcare-hq/pull/28409/files#diff-d28c016fb16e34a86ad9ce844e326789L622-L628

You could then remove `get_tracked_objects_to_create` and `settings.XFORM_TRACKED_MODELS`

Reply:
https://github.com/dimagi/commcare-hq/pull/28409/commits/f6728045d58f0b943e2693282aa8b8cb2e5b542b

Although I am bit hesitant to modify this already complex business logic. Earlier it was picking on specific tracked models but now it would pick on any model that is added on form and possibly call save again? can you confirm there is no such risk?